packages/account/src/predicate/predicate.ts

```diff
@@ -195,6 +196,20 @@ export class Predicate<TInputData extends InputValue[]> extends Account {
     }));
   }
 
+  /**
+   * Generates an array of fake resources based on the provided coins.
+   *
```

Comment:
Likewise, then it should be added to [methods](https://github.com/FuelLabs/fuels-ts/pull/2345/files#diff-8f528b24de2456d5421a8cd534e5d9d63d73c009665a62461f5c9565aa6b72ea)

Reply:
@maschad good point. I will add some docs for this.

Reply:
@maschad I've decided to not add a documentation page for this.

I remembered that some users had to learn how to create fake resources on their own to execute dry-run on contract calls to get value from the node. Since these contract calls were merely to read data, there was no need to waste resources submitting a real transaction. 

In the past, we did not provide a straightforward way to do so but this was addressed on https://github.com/FuelLabs/fuels-ts/pull/1860

Since I cannot think of another use case where users would want to generate fake resources on their own, I don't see much value in a documentation page for this since there won't be a real problem to solve.

However, this method is relevant for the changes that are going to be made [here](https://github.com/FuelLabs/fuels-ts/issues/2362), as predicate resources will be properly generated, without the need to be injected later on.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2351/commits/33c46397835fa686a17407de41f2a28f885213ff